app/main.py

```
@@ -96,15 +101,41 @@ def get_cir_metadata(
         for metadata in metadata_collection
         if not (
             (survey_id and metadata.survey_id != survey_id)
-            or (form_type and metadata.form_type != form_type)
+            or (classifier_type and metadata.classifier_type != classifier_type)
+            or (classifier_value and metadata.classifier_value != classifier_value)
             or (language and metadata.language != language)
-            or (status and metadata.status != status)
         )
     ]:
         return filtered_metadata
     raise HTTPException(status_code=404)
 
 
+@lru_cache(maxsize=1)
+def get_instrument_metadata(guid: UUID) -> CiMetadata:
+    """Return the CiMetadata object for the given guid, or raise HTTPException if not found"""
+    for schema in SCHEMAS_PATH.rglob("*.json"):
+        schema_json = json.loads(schema.read_text())
+        generated_guid = generate_guid_for_schema(schema.stem, schema_json["language"])
+        if generated_guid == guid:
+            return get_ci_metadata(guid, schema.stem, schema_json)
+    raise HTTPException(status_code=404, detail="Metadata not found for the given guid")
+
+
+@app.get("/v3/ci_metadata")
+def get_cir_metadata_v3(
+    guid: UUID,
+) -> dict:
+    """
+    Return the metadata object for the given guid as a dict.
+    Raises not found if no metadata can be found for the given guid.
+    """
+    try:
+        metadata = get_instrument_metadata(guid)
+        return metadata.model_dump()
+    except HTTPException as e:
```

Do we want to raise an exception with the status code and details, as above? Or at least:
```python
raise HTTPException(status_code=404)
```

---

Is that needed? The endpoint calls get_instrument_metadata(guid), and that throws a 404 if there is no match to the guid

---

So, I guess we never hit the except here? No need for try/except since it is handled in `get_instrument_metadata` and the function could be just:
```python
@app.get("/v3/ci_metadata")
def get_cir_metadata_v3(
    guid: UUID,
) -> dict[str, str | int | UUID]:
    """
    Return the metadata object for the given guid as a dict.
    Raises not found if no metadata can be found for the given guid.
    """
    metadata = get_instrument_metadata(guid)
    return metadata.model_dump()
```

---

Good point - change made